fix(gerrit): add codeowners support for gerrit

jutzig opened the pull request:
## Changes

Fixes CODEOWNERS support for the gerrit platform

## Context

I tried using CODEOWNERS in gerrit to automatically add reviewers, but found out it's not working. I came to the conclusion, that the problem is in code-owner#getBranchFiles, because it uses the sourceBranch of the Pr to diff.
With Gerrit, the value in sourceBranch isn't a real branch, so the diff fails. To fix that, I use the commit sha as a diff base for the gerrit platform.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

jutzig committed:
add codeowners support for gerrit

when a Pr object is created from a gerrit change, the sourceBranch in
the Pr points to the branch name renovate uses internally. In the case
of gerrit, this isn't a branch at all (just a a message in the gerrit
change).
During code-owners analysis, the code tries to use the sourceBranch of
the Pr to check what has changed with getBranchFiles.
Since the sourceBranch does not exist, this fails and yields no results.
As a fix code-owners now checks if it's the gerrit platform and if so,
diffs the Pr commit sha instead of the sourceBranch

jutzig committed:
adds new getBranchFilesFromRef method to avoid code duplication

rarkins committed:
fix lint

jutzig commented on lib/workers/repository/update/pr/code-owners.ts:
```diff
@@ -98,9 +99,14 @@ export async function codeOwnersForPr(pr: Pr): Promise<string[]> {
 
     // Get list of modified files in PR
-    const prFiles = await getBranchFiles(pr.sourceBranch);
+    //on the gerrit platform, the source branch does not exists. Instead of the branch, we can take the commit sha and diff against the parent
+    const prFiles =
+      'gerrit' === pl && pr.sha
+        ? await getBranchFilesFromCommit(pr.sha)
+        : await getBranchFiles(pr.sourceBranch);
```
@felipecrs thank you very much for taking the time to review!

> I don't think this needs to be platform specific.

I wanted to change as little as possible, but if it makes sense in general... sure

felipecrs replied:
You did good, let's just wait for @rarkins to confirm the way forward.

rarkins replied:
Yes, we prefer to avoid platform-specific code in workers

felipecrs replied:
That's exactly what I meant. :)

jutzig committed:
remove platform specific code in worker

if a Pr contains a sha, always use that to compute a diff.
Otherwise fall back to comparing branches as before

jutzig committed:
fix review comments

jutzig commented:
anything left open for me to change? Thanks!

jutzig commented:
@viceice I think your review is the last thing missing. Is there anything you want changed?
